File: test/resources/HelloProphecy/prophecy/pipelines/gold_sales/code/gold_sales/graph/SumAmounts.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.utils import *
from prophecy.libs import typed_lit
from gold_sales.config.ConfigStore import *
from gold_sales.udfs.UDFs import *

def SumAmounts(spark: SparkSession, in0: DataFrame) -> DataFrame:
    df1 = in0.groupBy(col("customer_id"))

    return df1.agg(count(col("order_id")).alias("orders"), sum(col("amount")).alias("total_spend"))

File: test/resources/HelloProphecy/prophecy_scala/pipelines/gold_sales/code/src/main/scala/io/prophecy/pipelines/gold_sales/graph/SumAmounts.scala
package io.prophecy.pipelines.gold_sales.graph

import io.prophecy.libs._
import io.prophecy.pipelines.gold_sales.udfs.PipelineInitCode._
import io.prophecy.pipelines.gold_sales.udfs.UDFs._
import io.prophecy.pipelines.gold_sales.config.Context
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object SumAmounts {

  def apply(context: Context, in: DataFrame): DataFrame =
    in.groupBy(col("customer_id"))
      .agg(count(col("order_id")).as("orders"),
           sum(col("amount")).as("total_spend")
      )

}
